wallet/src/nightfall-browser/services/database.js

```diff
@@ -268,3 +270,81 @@ export async function markWithdrawState(transactionHash, withdrawState) {
     tx._id,
   );
 }
+
+/*
+Functions for manipulating the sibling path collection
+*/
+export async function getAllSiblingPaths() {
+  const db = await connectDB();
+  const commitments = await db.getAllKeys(SIBLING_COLLECTION);
+  return Promise.all(
+    commitments.map(async c => {
```

Comment:
This seems to me to iterate over the collection to get the sibling paths.  Would not an atomic `find()` to return an array of `siblingPaths` be more efficient?

Reply:
It would but I don't know of a `find-like` call in indexedDB.

The reason I have to do a `getAllKeys` and iterate is that the indexedDB schema for this collection is keyed by commitmentHash. Unfortunately, `get` in indexedDB doesn't also return the key, just the value. As I need the commitmentHash as part of the return object I think I'm forced to iterate (so I can re-pair up the commitmentHash to the corresponding stored values).

Alternatively I can duplicate commitmentHash in the key and value stored - but that violates DRY.

Reply:
Fair enough